### ISSUE TITLE
API addition to allow 3rd party (tt-rss) integration in 1.9.1

### ISSUE THREAD

**Issue by joshp23** [wallabag/wallabag#1425]
Hey guys,

Any way to get this included into Wallabag? 

TT-RSS is a self-hosted, online rss feed aggregator, similar (but superior IMO) to the old Google Reader application.

A plugin dev over on TT-RSS forums managed to make a plugin for posting RSS feed items in TT-RSS directly to a Wallabag installation from within the TT-RSS application itself, but had to patch Wallabag to achieve this without an annoying auth popup.

[Here](https://tt-rss.org/forum/viewtopic.php?f=22&t=3009#p17674) is the forum post on tt-rss discussing this.

[Here](https://github.com/xppppp/wallabag) is the patched version of Wallabag on github.

[Here](https://github.com/wallabag/wallabag/compare/master...xppppp:master) is the comparison of the patched version against unpatched Wallabag, obviously just showing what the patch did.

Perhaps this would help solve the annoying auth pop-ups that occur with the various plugins, like the chrome plugin and the Android app?

Thanks for looking!

**Comment by tcitworld** [comment]
Hi, 
We know of this work. It was done a while ago, and attempted to provide an API for wallabag. But now, we have worked on a whole new version of wallabag that is much more clean and complete, and it features an API. So now TTRSS will be able to add items directly into wallabag using our new API.

**Comment by joshp23** [comment]
You know what, I even knew that... so this plugin and api patch works for the 1.9.x branch, but with 2.0 there is a whole new world of API awesomeness that you have implemented. That sound about right?

No more messing about before coffee and meditation. Check.

Thanks!

**Comment by tcitworld** [comment]
Totally right. Now we've got to find someone who is willing to create a plugin for TTRSS that works with our new API.

**Comment by joshp23** [comment]
Just wanted to add to this that the patch mentioned above is incomplete after testing. It seems to work fine on 1.9, but breaks on 1.9.1. I corrected this for folks on 1.9.1, like myself, and detailed the difference in the thread I mentioned above in [this post](https://tt-rss.org/forum/viewtopic.php?f=22&t=3009&p=20961#p20961). Folks on 1.9.1 that want this must use the instructions there.

So there is clear api functionality for TTRSS and Wallabag to communicate for 1.9 and 1.9.1 thanks to that patch effort, with a small bit of editing. Thanks for the great work you folks are doing on this project!

**Comment by tcitworld** [comment]
> It will allow you to post, but the auth wall still blocks you.

You may be interested by [this page](http://doc.wallabag.org/en/Administrator/sessions.html) about sessions.

> It would be great to see that patch realized for 1.9.1 while we wait for 2.0. *

I can't speak for everyone, but I'm not sure we would gain much time by doing this. v1.9.1 was intended to be the last 1.x version, and v2 starts to show up. 
As for TTRSS -> wallabag share, tt-rss-wallabag's plugin from Heiko Adams seems to do the trick, even though you can still encounter session issues (which should be fixed by the link above).

You are still free to open a PR so that we look things up, but don't get high hopes of it being merged.

**Comment by joshp23** [comment]
Hey, thanks for that, I will look into that session info post haste. I did manage to get xppppp to update his patch repo, and will be testing today. I will update here if it works out. Incidentally, his plugin does work out of the box for publishing to wallabag from ttrss, it just requires a manual auth, which is what the "crude api.php" is supposed to work around. At any rate, I look forward to 2.0!!!

**Comment by tcitworld** [comment]
>  At any rate, I look forward to 2.0!!!

We all do.
For all that would be interested in starting the work for a TTRSS->wallabag export plugin when v2's out, there's our documentation : http://v2.wallabag.org/api/doc#post--api-entries.{_format}

**Comment by joshp23** [comment]
I just wanted to close this off with a resounding cry for success. For any folks that stumble on this, to resolve the failure:
* I first followed the instructions on [this page](http://doc.wallabag.org/en/Administrator/sessions.html) that tcitworld pointed me to
* I then updated the patch files from the update in xppppp's repo fork. 
* Then I re-entered in the user and url info in the ttrss patch after entering in bogus info to reset things

and it works like a charm. Thank you!
